spec/feedx/task_state_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+RSpec.describe Feedx::TaskState do
+  subject { described_class }
+  after   { subject.clear }
+
+  it 'should run with state' do
+    subject.with('blank') do |state|
+      expect(state).to eq({})
+    end
+
+    subject.with('sums', default: { sum: 0 }) do |state|
+      expect(state).to eq(sum: 0)
+    end
+
+    rnd = Random.new(101)
+    10.times do
+      subject.with('sums') {|s| s[:sum] += rnd.rand(10) }
+    end
+    expect(subject['sums']).to eq(sum: 53)
```

Comment:
Oh. Are you sure? Probs, better just do a counter without anything random?

Reply:
yeah, that was just to demonstrate that you can use it with something that has its external state (like the seeded random). it's working fine